backend/src/game_server.rs

```diff
@@ -247,12 +247,29 @@ impl GameServer {
 
         self.deselect_unit();
 
-        self.send_current_player(State::new(STATE_WAIT.to_string()));
+        if self.game.ends() {
```

Comment:
you may notice that this solution is quite ugly

there was an idea about wrapping response messages into enum and choose one of enum instances in condition branch, but it requires quite a big refactoring

i think all system with sending messages should be refactor with this strange communicator.rs

Reply:
Better to write 3 lines of code instead of function